### PR TITLE
Support more recent Mill versions

### PR DESCRIPTION
**Pull request by lefou**

### CODE REVIEW

**Comment by lefou** [comment]
@lolgab MIll 0.10.0-M5 introduced an source-incompatible API change for `prepareOffline`. What's your prefered way to support this version?

1. Have a full-copy of `trait Mima` with version-specific differences source dirs, or
2. introduce a shared base trait with common code and only have the version-specific implementations in the `Mima` trait?

**Comment by lolgab** [comment]
Hi @lefou,
I already implement the shared trait strategy. I will push a commit to your branch if you don't mind. Ok?

**Comment by lolgab** [comment]
@lefou I don't know how to handle the `all` parameter in the new `prepareOffline`.
Also do you like the idea of `MillBase` ? Do you have a better name for it? It was failing to compile because of the cyclic dependency.

**Comment by lolgab** [comment]
Also, this (guess what 😄) breaks binary compatibility.. I'm thinking I'll maintain the `0.0.x` version scheme for a long long time 😆

**Comment by lefou** [comment]
Wow, you must have a huge count of projects depending on a stable API of Mima. Just kidding.

I'd rather move the final `Mima` trait in the version-specific source folder and remove the extra inherited offline-support traits.

**Comment by lolgab** [comment]
You mean to have copy-pasted code for everything?

**Comment by lefou** [comment]
I meant to have one shared `MimaBase` and two concrete but version-specific `Mima` that only hold the different `prepareOffline` defs.

**Comment by lolgab** [comment]
Ooh true, great idea 💡
Didn't think about that.

**Comment by lefou** [comment]
Regarding the `all` flag: it depends how you like to see the importance of your plugin in a day-to-day workflow. 

If you say, it's only relevant when making releases and creating pull requests, than you shouldn't fetch by default, so you should only fetch when the `all` flag is `true`. That would by also my personal preference.

**Comment by lolgab** [comment]
But do I have to check the flag before the `T.command` block? Otherwise it fetches the dependencies either way. Right?

**Comment by lefou** [comment]
Exactly! Good example is in `ScalaModule.prepareOffline` (0.10.0-M5)

File: mill-mima/src-mill0.10.0-M5/com/github/lolgab/mill/mima/Mima.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.github.lolgab.mill.mima
+
+import mill._
+import mill.define.Command
+import mill.scalalib._
+
+trait Mima extends MimaBase with OfflineSupportModule {
```
Comment (lefou) [inline mill-mima/src-mill0.10.0-M5/com/github/lolgab/mill/mima/Mima.scala:7]: Just denoting, that `with OfflineSupportModule` is just for documentation, as `ScalaModule` is already a `OfflineSupportModule`.

**Review by lefou — COMMENTED** [review]
LGTM!

**Pull request merged**